src/main/groovy/com/bugsnag/android/gradle/BugsnagPluginExtension.java

```diff
@@ -0,0 +1,148 @@
+package com.bugsnag.android.gradle;
+
+import java.util.Map;
+
+/**
+ * Defines configuration options (Gradle plugin extensions) for the BugsnagPlugin
+ */
+public class BugsnagPluginExtension {
+
+    String endpoint = "https://upload.bugsnag.com";
+    String releasesEndpoint = "https://build.bugsnag.com";
+
+    boolean autoUpload = true;
+    boolean autoReportBuilds = true;
+    boolean uploadDebugBuildMappings = false;
+    boolean overwrite = false;
+    int retryCount = 0;
+    Boolean ndk = null;
+    String sharedObjectPath = null;
+    String projectRoot = null;
+    boolean failOnUploadError = true;
+    int requestTimeoutMs = 60000;
+
+    // release API values
+    String builderName = null;
+    Map<String, String> metadata = null;
+    Map<String, String> objdumpPaths = null;
```

Comment:
If these properties aren't private, do we need the getters & setters?

Reply:
I believe these are required for interop with Groovy. I attempted leaving these without the getters/setters previously but this resulted in a compilation error in the NDK E2E tests because the `bugsnag.ndk` property couldn't be found. Adding accessors resolved this issue.